sites/svelte-5-preview/src/routes/docs/content/01-api/02-runes.md

```diff
@@ -561,6 +561,26 @@ $inspect(stuff).with(console.trace);
 
 > `$inspect` only works during development.
 
+## `$host`
+
+Retrieves the `this` reference of the custom element that contains this component. Example:
+
+```svelte
+<svelte:options customElement="my-element" />
+
+<script>
+	function greet(greeting) {
+		$host().dispatchEvent(
+			new CustomEvent('greeting', { detail: greeting })
+		);
+	}
+</script>
+
+<button onclick={() => greet('hello')}>say hello</button>
+```
+
+> Only available inside custom element components, and only on the client-side
```

Comment:
> Only available within client-side custom element components.

Reply:
This raises an interesting question — what even _is_ a server-side custom element component? What is supposed to get rendered? Should we be emitting declarative shadow DOM?

Are there any situations where `$host()` could be reached in a server environment? Instead of `undefined`, should it be a function call that throws a useful error (perhaps only in dev?)

Reply:
Calling `$host()` doesn't necessarily mean that you access one of its properties right away. You could save this to a variable. So an error could have false positives.

Reply:
What about a proxy that threw on property access? That way if someone did something like 

```js
const parent = $host().parentNode;
```

they would get a more useful error than `Cannot read properties of undefined (reading 'parentNode')` or whatever

Reply:
That could have intended consequences in the case of checking if host is defined when writing code that is eagerly executed on the server